zeebe/zb-db/src/main/java/io/camunda/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java

```diff
@@ -48,26 +48,34 @@ public final class ZeebeRocksDbFactory<
     RocksDB.loadLibrary();
   }
 
+  private final int partitionId;
   private final RocksDbConfiguration rocksDbConfiguration;
   private final ConsistencyChecksSettings consistencyChecksSettings;
+  private final boolean enableAccessMetrics;
 
   public ZeebeRocksDbFactory(
+      final int partitionId,
       final RocksDbConfiguration rocksDbConfiguration,
-      final ConsistencyChecksSettings consistencyChecksSettings) {
+      final ConsistencyChecksSettings consistencyChecksSettings,
+      final boolean enableAccessMetrics) {
+    this.partitionId = partitionId;
     this.rocksDbConfiguration = Objects.requireNonNull(rocksDbConfiguration);
     this.consistencyChecksSettings = Objects.requireNonNull(consistencyChecksSettings);
+    this.enableAccessMetrics = enableAccessMetrics;
   }
 
   @Override
   public ZeebeTransactionDb<ColumnFamilyType> createDb(final File pathName) {
     final List<AutoCloseable> closeables = Collections.synchronizedList(new ArrayList<>());
     try {
       return ZeebeTransactionDb.openTransactionalDb(
+          partitionId,
           prepareOptions(closeables),
           pathName.getAbsolutePath(),
           closeables,
           rocksDbConfiguration,
-          consistencyChecksSettings);
+          consistencyChecksSettings,
+          enableAccessMetrics);
```

Comment:
Alternative proposal would from my side to pass in a metrics implementation, so the API accepts some metrics interface.

This way we could better test it due to DI and we wouldn't need to pass in the partition ID (I see this a bit like polluting the api as the db shouldn't care about partitions.)

Furthermore, we could have a default implementation which is noop. 


What do you think?